In [1]:
import pandas as pd

# 데이터 불러오기
file_path = '../data/drinks.csv'
drinks = pd.read_csv(file_path) # read_csv 함수로 데이터를 Dataframe 형태로 불러옵니다.

# 유럽 국가들의 데이터 추출
eu_drinks = drinks.loc[drinks['continent'] == 'EU']

# 맥주와 와인 간의 상관관계 계산
corr = eu_drinks['beer_servings'].corr(eu_drinks['wine_servings'], method='pearson')

print('맥주와 와인 간의 피어슨 상관값:', corr)

맥주와 와인 간의 피어슨 상관값: 0.2977739788485701


문제 : 유럽국가들에 대해서는 맥주와 와인간의 상관관계는 얼마나 되는지 피어슨상관값(Pearson Correlation)을 구하기

음주데이터를 불러오고,
유럽국가들에 대하여 구해야하니 유럽 국가의 데이터를 우선 추출을 한다.
그 데이터 중에서 맥주 소비량과 와인 소비량을 추출해 바로 피어슨 상관계산을 하였다.
피어슨 상관값이 0.2978로 양의 선형 상관관계가 있다고 판단할 수 있고,
맥주 소비량이 증가할 때 와인 소비량도 약간 증가하는 경향이 있을 수 있다고 판단하였다.